sklego/pandas.py

```diff
@@ -0,0 +1,95 @@
+import pandas as pd
+
+
+def negate_lags(lags):
+    """
+    Helper function, negates an int or list of ints.
+
+    >>> negate_lags(3)
+    [-3]
+
+    >>> negate_lags([3])
+    [-3]
+    """
+
+    def negate(x):
+        return -x
+
+    if isinstance(lags, int):
+        return [negate(lags)]
+
+    if hasattr(lags, '__iter__'):
+        assert all(isinstance(x, int) for x in lags), "Must be list of ints"
+        return list(map(negate, lags))
+
+
+def expand_columns(cols):
+    """
+    Helper function, returns list of one or more columns
+    from column or list of columns.
+
+    >>> expand_columns('test')
+    ['test']
+
+    >>> expand_columns(['test1', 'test2'])
+    ['test1', 'test2']
+    """
+
+    if isinstance(cols, str):
+        return [cols]
+
+    if hasattr(cols, '__iter__'):
+        return cols
+
+
+def add_lag(X, cols, lags):
+    """
```

Comment:
might be good to check if `X` is indeed a pandas dataframe or a numpy array before doing anything and returning an value error if this isnt the case. 

same holds for if the index or the column is missing. having a valuerror from our library is better than waiting for later packages. it's more friendly to uders

Reply:
also. params!